spec/services/purchase_service_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+RSpec.describe PurchaseService, type: :model do
+  let(:address) { build(:address) }
+  let(:valid_user) { create(:user) }
+  let(:cart) { create(:cart, user: valid_user) }
+  subject {{gateway: "paypal", cart_id: cart.id, address: address}}
+
+  context 'When parameters are valid' do
+    it "returns order instance" do
+      expect(PurchaseService.call(subject)[:success]).to be_truthy
+    end
+  end
+
+  it "returns error message if gateway is invalid" do
```

Comment:
As vezes você usa aspas simples e as vezes duplas. No geral a comunidade usa simples a não ser que esteja concatenando strings com `"My name is #{name}"`